### PR TITLE
rfc: update documentation of new versions/image API

### PR DESCRIPTION
**Pull request by katexochen**
Changes to include: https://gist.github.com/katexochen/4f8a51117b8d68c0dcd880ac51c8a41b

### CODE REVIEW

File: rfc/apis.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+- `ref/-/stream/<stream>/<version>` is equivalent to the short form `stream/<stream>/<version>`
+
+
+## Resource names
+
+Resources can be identified by `ref/<ref>/stream/<stream>/<version>` and allow for short form encoding as explained above.
+Resources using this encoding include:
+
+- `version`: version information for Constellation components
+- `image`: metadata for individual Constellation OS images
```
Comment (katexochen) [inline rfc/apis.md:71]: This part isn't really clear yet. Also, it doesn't fit well the API paths...

File: rfc/apis.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  - `sbom`: SBOMs for Constellation OS images
+
+There may be more API groups in the future (e.g. `cli`)
+
+## API paths overview
+
+- [`/constellation/v1/ref/<ref>/stream/<stream>/versions/latest/<kind>.json`](version-api.md#latest)
+- [`/constellation/v1/ref/<ref>/stream/<stream>/versions/major/<base>/<kind>.json`](version-api.md#major-to-minor-version-list)
+- [`/constellation/v1/ref/<ref>/stream/<stream>/versions/minor/<base>/<kind>.json`](version-api.md#minor-to-patch-version-list)
+- [`/constellation/v1/ref/<ref>/stream/<stream>/image/<version>/info.json`](image-api.md#image-lookup-table)
```
Comment (katexochen) [inline rfc/apis.md:30]:
```suggestion
- [`/constellation/v1/ref/<ref>/stream/<stream>/<version>/image/info.json`](image-api.md#image-lookup-table)
```
Might be better the other way around, especially regarding the meaning of a short path.

**Comment by katexochen** [comment]
We should also define how to discover major versions in the versions API. Currently we can just assume it is v2, but be should consider that we might want to bump the major version and might also have tooling that in this case would need to deal with both major vers (i.e., tooling that couldn't use a constant string anymore).

File: rfc/apis.md (hunk 2 above)
Comment (katexochen) [reply rfc/apis.md:30]: This would also make deleting a single version much simpler.

File: rfc/apis.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+- `debug`: Image containing the debugd, allows access to the serial console
+
+## Short paths for resource identification
+
+Short paths allow an easier version handling, e.g., in end user configuration files or CLI tools.
+When using the default values, short paths omit either the `ref` (if set to `-`) or the `ref` and `stream`
+parts (if set to `-` and `stable`):
+
+- `ref/-/stream/stable/<version>` is equivalent to the short path `<version>`
+- `ref/-/stream/<stream>/<version>` is equivalent to the short path `stream/<stream>/<version>`
```
Comment (katexochen) [inline rfc/apis.md:92]: How would a short path for API version v2 look like? If the API is versioned but users are using an unversioned short path, much of the benefits of the versioning seems to be lost.

Comment (malt3) [reply rfc/apis.md:92]: The API version should probably be hardcoded in the API client and not be part of the short name. Instead the short name should stay forwards compatible with new API versions.

File: rfc/apis.md (hunk 2 above)
Comment (katexochen) [reply rfc/apis.md:30]: As discussed with @malt3 , I will implement this change before the next release.

File: rfc/version-api.md [hunk 4]
```diff
@@ -108,13 +121,18 @@ The CLI can query the `latest` endpoint to inform users about new releases of th
...
 
-- Explicit compatbility information between different resource kinds
+- Explicit compatibility information between different resource kinds
     - Example: OS image `v2.3.4` requires microservice version > `v2.3.1`
     - Allows fine grained control in case of unexpected incompatibility
 - Version metadata
     - Changelogs
     - Deprecation warnings
-    - Security information
+    - Security informatio
```
Comment (malt3) [inline rfc/version-api.md:138]:
```suggestion
    - Security information
```

**Review by malt3 — APPROVED** [review]
otherwise lgtm

**Pull request merged**